example-app/src/main/java/com/adyen/checkout/example/ui/card/CardViewModel.kt

```diff
@@ -85,7 +83,7 @@ internal class CardViewModel @Inject constructor(
         when {
             state == null -> _cardViewState.tryEmit(CardViewState.Error)
             state.isValid -> makePayment(state.data)
-            else -> _cardViewState.tryEmit(CardViewState.Invalid)
+            else -> viewModelScope.launch { _events.emit(CardEvent.Invalid) }
```

Comment:
If we specify a buffer for this `SharedFlow` then `tryEmit` would work. This will also work, but I can change it to be more consistent with other event like flows. WDYT?

Reply:
No need to, it's just the example app 😉